packages/liquid-html-parser/src/stage-1-cst.ts

```diff
@@ -111,11 +112,18 @@ export interface ConcreteBasicNode<T> {
 export interface ConcreteLiquidDocParamNode
   extends ConcreteBasicNode<ConcreteNodeTypes.LiquidDocParamNode> {
   name: 'param';
-  paramName: ConcreteTextNode;
+  paramName: ConcreteLiquidDocParamNameNode;
   paramDescription: ConcreteTextNode | null;
   paramType: ConcreteTextNode | null;
 }
 
+export interface ConcreteLiquidDocParamNameNode
```

Comment:
It may look kind of weird to have `required` live in this node, however this is consistent with how the parsing logic works, and allows us to use the `mappings` in `stage-1`.

We do this because it aligns with the parsing grammar, and allows us to leverage the `Mappings` mechanism. We map this to a more sensible location in `stage-2` [here](https://github.com/Shopify/theme-tools/pull/733/files#diff-d4d621a0e25c584008f71bf2ca34d635d50e0a6a1af987addb4ecf76ce649300R1302) (`node.paramName.required -> node.required`)